src/lib.rs

```diff
@@ -33,26 +33,25 @@ extern crate alloc;
 
 extern crate bitcoin_hashes;
 
-#[cfg(feature = "std")]
+#[cfg(any(feature = "alloc", feature = "std"))]
 extern crate unicode_normalization;
 
-#[cfg(feature = "rand_core")]
-pub extern crate rand_core;
 #[cfg(feature = "rand")]
 pub extern crate crate_rand as rand;
+#[cfg(feature = "rand_core")]
+pub extern crate rand_core;
 #[cfg(feature = "serde")]
 pub extern crate serde;
 
+use alloc::{borrow::Cow, string::ToString, vec::Vec};
```

Comment:
I'm surprised this isn't feature gated on the "alloc" feature?

Reply:
Fixed.